lib/matplotlib/tests/test_dates.py

```diff
@@ -511,6 +511,162 @@ def _create_auto_date_locator(date1, date2):
+    'day: 05',
+    'day: 09',
+    'day: 13',
+    'day: 17',
+    'day: 21',
+    'day: 25',
+    'day: 29',
+    '02/1997',
+    'day: 05',
+    'day: 09']],
+    [datetime.timedelta(hours=40),
+    ['day: 01',
+    '04 hr 00 min',
+    '08 hr 00 min',
+    '12 hr 00 min',
+    '16 hr 00 min',
+    '20 hr 00 min',
+    'day: 02',
+    '04 hr 00 min',
+    '08 hr 00 min',
+    '12 hr 00 min',
+    '16 hr 00 min']],
+    [datetime.timedelta(minutes=20),
+    ['00 hr 00 min',
+    '00 hr 05 min',
+    '00 hr 10 min',
+    '00 hr 15 min',
+    '00 hr 20 min']],
+    [datetime.timedelta(seconds=40),
+    ['00 hr 00 min',
+    '05.000000 sec',
+    '10.000000 sec',
+    '15.000000 sec',
+    '20.000000 sec',
+    '25.000000 sec',
+    '30.000000 sec',
+    '35.000000 sec',
+    '40.000000 sec']],
+    [datetime.timedelta(seconds=2),
+    ['59.500000 sec',
+    '00 hr 00 min',
+    '00.500000 sec',
+    '01.000000 sec',
+    '01.500000 sec',
+    '02.000000 sec',
+    '02.500000 sec']],
+     )
+    for t_delta, expected in results:
+        d2 = d1 + t_delta
+        strings = _create_auto_date_locator(d1, d2)
+        assert strings == expected
+
+
+def test_concise_formatter_zformats():
+    zero_formats = ['', "'%y", '%B', '%m-%d', '%S', '.%f']
+
+    def _create_auto_date_locator(date1, date2):
+        fig, ax = plt.subplots()
+
+        locator = mdates.AutoDateLocator(interval_multiples=True)
+        formatter = mdates.ConciseDateFormatter(
+            locator, zero_formats=zero_formats)
+        ax.yaxis.set_major_locator(locator)
+        ax.yaxis.set_major_formatter(formatter)
+        ax.set_ylim(date1, date2)
+        fig.canvas.draw()
+        sts = []
+        for st in ax.get_yticklabels():
+            sts += [st.get_text()]
+        return sts
+
+    d1 = datetime.datetime(1997, 1, 1)
+    results = ([datetime.timedelta(weeks=52 * 200),
+                [str(t) for t in range(1980, 2201, 20)]
+                ],
+               [datetime.timedelta(weeks=52),
+                ["'97", 'Feb', 'Mar', 'Apr', 'May', 'Jun',
+                    'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+                ],
+               [datetime.timedelta(days=141),
+                ['January', '22', 'February', '22', 'March',
+                    '22', 'April', '22', 'May', '22']
+                ],
+               [datetime.timedelta(days=40),
+                ['January', '05', '09', '13', '17', '21',
+                    '25', '29', 'February', '05', '09']
+                ],
+               [datetime.timedelta(hours=40),
+                ['01-01', '04:00', '08:00', '12:00', '16:00', '20:00',
+                    '01-02', '04:00', '08:00', '12:00', '16:00']
+                ],
+               [datetime.timedelta(minutes=20),
+                ['00', '00:05', '00:10', '00:15', '00:20']
+                ],
+               [datetime.timedelta(seconds=40),
+                ['', '05', '10', '15', '20', '25', '30', '35', '40']
+                ],
```

Comment:
OK, these results are a little confusing, I suspect because you are starting 1 Jan?  Does it make sense to start on a different date and a bit later in the day?

Reply:
I think the issue is that %S is not zero-padded, perhaps if I switch the formatting string to '%s' which is zero padded, then the string will not be empty and the result less confusing?

Reply:
I made the last element in zero_formats "%S.%f' in stead of ".%f" to make the expected string '00' instead of the empty string above.

Reply:
I think its a bit confusing to have the zero-format for seconds be seconds, but I guess this indeed tests the functionality.